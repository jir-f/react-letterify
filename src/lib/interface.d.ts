interface LetterifyProps {
  letterifyString: string,
  styleClasses?: string[],
  animate: boolean,
  random?: boolean,
  delay?: number,
  speed?: number,
  direction?: string,
  color?: string
}

interface LetterifyState {
  opacity: number,
  direction: string,
  speed: number,
  delay: number,
}