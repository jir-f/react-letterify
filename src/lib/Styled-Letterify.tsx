import styled, {keyframes} from 'styled-components';

export const LetterifyStyle = styled.span`
	color: ${props => props.theme.color};
	display: inline-block;
	font-weight: bold;
	margin-right: 5px;
	opacity: ${props => props.theme.attrs.opacity};
	transform: ${props => `translate(${props.theme.attrs.xValue}px, ${props.theme.attrs.yValue}px)`};
	transition: ${props => `all ${props.theme.timing}s ease ${props.theme.delay}s`}
`;