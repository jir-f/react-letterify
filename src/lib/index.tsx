export { default as Letterify } from './letterify/Letterify';