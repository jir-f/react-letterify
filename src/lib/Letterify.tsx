import React, { Component } from 'react';
import { LetterifyStyle } from './Styled-Letterify';

/// <reference path="./interface.d.ts"/>

export default class Letterify extends Component<LetterifyProps, LetterifyState>{
	constructor(props: LetterifyProps) {
		super(props);
		this.state={opacity: 0,
			direction: 'inplace',
			speed: 5,
			delay: 0.1
		};
	}

	componentDidMount(){
		console.log(this.props);
	}

	static getDerivedStateFromProps(nextProps: LetterifyProps){
		if(nextProps.animate){
			return ({opacity: 1});
		} else {
			return ({opacity: 0,});
		}
	}

	letterify = (passedSentence: string) => {
		if(passedSentence && passedSentence.length > 0){
			return(
				<span className='letterify__word'>
					{this.splitLetters(passedSentence)}
				</span>
			);
		}
	}

	splitLetters = (passedSentence: string)=>{
		const result: JSX.Element[] = [];
		var i: number = 1;
		const translateValue: any = 'inplace';

		for (var c of passedSentence) {
			const t = {timing: 0.2, 
				delay: i/10, 
				opacity: this.state.opacity,
				translateX: translateValue.x,
				translateY: translateValue.y
				}
			result.push(<LetterifyStyle theme={t} className='letterify__word_char' key={i = i+1}>{c}</LetterifyStyle>)
		}
		return result;
	}
	
	getLetterifyClasses = () => {
		if(this.props.styleClasses){
			var result: string = '';
			const spacing: string = this.props.styleClasses.length > 1 ? ' ' : '';
			this.props.styleClasses.forEach((item) => {
				return result = result + item + spacing;
			});
			return result;
		}
	}

	render(){
		return(
			<div className={this.getLetterifyClasses()}>
				{this.letterify(this.props.letterifyString)}
			</div>
		);
	}
}