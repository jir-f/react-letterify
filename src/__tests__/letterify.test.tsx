import React from 'react';
import { Letterify } from '../lib';
import renderer from 'react-test-renderer';

describe('Letterfiy tests', () => {
  it('animates letters direction inplace', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
      letterifyString="Test letterify component"
      animate={animate}
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

  it('animates letters direction up', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
      letterifyString="Test letterify component"
      animate={animate}
      direction={"up"}
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

  it('animates letters direction down', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
      letterifyString="Test letterify component"
      animate={animate}
      direction={"down"}
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

  it('animates letters direction right', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
      letterifyString="Test letterify component"
      animate={animate}
      direction={"right"}
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

  it('animates letters direction left', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
      letterifyString="Test letterify component"
      animate={animate}
      direction={"left"}
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

  /* Mising props */
  it('missing letterfiy string', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
      animate={animate}
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

  it('missing animate boolean', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
      letterifyString="Test letterify component"
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

  it('missing all reuired props', () => {
    var animate: boolean = false;
    const letteriftTree = renderer
    .create(<Letterify 
    />)
    .toJSON();
  expect(letteriftTree).toMatchSnapshot();
  });

});