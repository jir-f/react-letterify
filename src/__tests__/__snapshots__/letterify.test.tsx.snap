// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letterfiy tests animates letters direction down 1`] = `
<div
  className="no__styleclass"
>
  <span
    className="letterify__word"
  >
    <span
      className="letterify__word_char sc-bdVaJa hcTaWg"
    >
      T
    </span>
    <span
      className="letterify__word_char sc-bdVaJa gOUnmE"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dabUSe"
    >
      s
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hMnOzQ"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eCMijJ"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa gYAKYU"
    >
      l
    </span>
    <span
      className="letterify__word_char sc-bdVaJa efTlbf"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eGQHsI"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jrephA"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa laPWEY"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bZhrAE"
    >
      r
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cHWCqC"
    >
      i
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bKOdGn"
    >
      f
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dhihQU"
    >
      y
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fvfKLt"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bHhQrv"
    >
      c
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dtTVSS"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hHqYqQ"
    >
      m
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bqJSiK"
    >
      p
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hjbgGG"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eVODNt"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa exxOcb"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa iMtjQg"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa blcidZ"
    >
      t
    </span>
  </span>
</div>
`;

exports[`Letterfiy tests animates letters direction inplace 1`] = `
<div
  className="no__styleclass"
>
  <span
    className="letterify__word"
  >
    <span
      className="letterify__word_char sc-bdVaJa cLaLDe"
    >
      T
    </span>
    <span
      className="letterify__word_char sc-bdVaJa itTzVI"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fqcpQM"
    >
      s
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bhzkWh"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hSnJnP"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cYMYvq"
    >
      l
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cfSahW"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dcDfss"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa xFlAG"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hqOdAF"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa RkTM"
    >
      r
    </span>
    <span
      className="letterify__word_char sc-bdVaJa gkjZmh"
    >
      i
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hmUMwK"
    >
      f
    </span>
    <span
      className="letterify__word_char sc-bdVaJa lJqJa"
    >
      y
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fWBUwy"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dTaxIf"
    >
      c
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cSZuIy"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jamlIG"
    >
      m
    </span>
    <span
      className="letterify__word_char sc-bdVaJa gvgRZf"
    >
      p
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hTsBGE"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa djNknU"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eGnfTS"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jETsEM"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jbSYzh"
    >
      t
    </span>
  </span>
</div>
`;

exports[`Letterfiy tests animates letters direction left 1`] = `
<div
  className="no__styleclass"
>
  <span
    className="letterify__word"
  >
    <span
      className="letterify__word_char sc-bdVaJa hMHHWj"
    >
      T
    </span>
    <span
      className="letterify__word_char sc-bdVaJa kgUgcZ"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jFiffE"
    >
      s
    </span>
    <span
      className="letterify__word_char sc-bdVaJa kQwmkA"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa iqopFy"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa tVqBu"
    >
      l
    </span>
    <span
      className="letterify__word_char sc-bdVaJa iwFofl"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bjIpDa"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa lhJmIf"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jpJhEQ"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa ctzNEF"
    >
      r
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eAEWku"
    >
      i
    </span>
    <span
      className="letterify__word_char sc-bdVaJa diMUnC"
    >
      f
    </span>
    <span
      className="letterify__word_char sc-bdVaJa ccowdq"
    >
      y
    </span>
    <span
      className="letterify__word_char sc-bdVaJa edYkLV"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bYhoTR"
    >
      c
    </span>
    <span
      className="letterify__word_char sc-bdVaJa havoOp"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa djaScs"
    >
      m
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jlsQOz"
    >
      p
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cgzJzX"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa ibiyfb"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa NdcOL"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dKqaGE"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dkPaJU"
    >
      t
    </span>
  </span>
</div>
`;

exports[`Letterfiy tests animates letters direction right 1`] = `
<div
  className="no__styleclass"
>
  <span
    className="letterify__word"
  >
    <span
      className="letterify__word_char sc-bdVaJa hMHHWj"
    >
      T
    </span>
    <span
      className="letterify__word_char sc-bdVaJa kgUgcZ"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jFiffE"
    >
      s
    </span>
    <span
      className="letterify__word_char sc-bdVaJa kQwmkA"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa iqopFy"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa tVqBu"
    >
      l
    </span>
    <span
      className="letterify__word_char sc-bdVaJa iwFofl"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bjIpDa"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa lhJmIf"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jpJhEQ"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa ctzNEF"
    >
      r
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eAEWku"
    >
      i
    </span>
    <span
      className="letterify__word_char sc-bdVaJa diMUnC"
    >
      f
    </span>
    <span
      className="letterify__word_char sc-bdVaJa ccowdq"
    >
      y
    </span>
    <span
      className="letterify__word_char sc-bdVaJa edYkLV"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bYhoTR"
    >
      c
    </span>
    <span
      className="letterify__word_char sc-bdVaJa havoOp"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa djaScs"
    >
      m
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jlsQOz"
    >
      p
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cgzJzX"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa ibiyfb"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa NdcOL"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dKqaGE"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dkPaJU"
    >
      t
    </span>
  </span>
</div>
`;

exports[`Letterfiy tests animates letters direction up 1`] = `
<div
  className="no__styleclass"
>
  <span
    className="letterify__word"
  >
    <span
      className="letterify__word_char sc-bdVaJa jijbax"
    >
      T
    </span>
    <span
      className="letterify__word_char sc-bdVaJa equCyu"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa flXpaj"
    >
      s
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fAcXlv"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cHEUzU"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cPjnlW"
    >
      l
    </span>
    <span
      className="letterify__word_char sc-bdVaJa iiRvko"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jvvUzB"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eRhRdp"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dUbvBr"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa MNXvU"
    >
      r
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jHCDRG"
    >
      i
    </span>
    <span
      className="letterify__word_char sc-bdVaJa Nsnqe"
    >
      f
    </span>
    <span
      className="letterify__word_char sc-bdVaJa xFtUa"
    >
      y
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jlqkSF"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa pkpIe"
    >
      c
    </span>
    <span
      className="letterify__word_char sc-bdVaJa gXcryB"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jAThhg"
    >
      m
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jaVYMB"
    >
      p
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cTOzYa"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fzWWqL"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fYJiut"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fjRwVt"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa PLuVY"
    >
      t
    </span>
  </span>
</div>
`;

exports[`Letterfiy tests missing all reuired props 1`] = `
<div
  className="no__styleclass"
/>
`;

exports[`Letterfiy tests missing animate boolean 1`] = `
<div
  className="no__styleclass"
>
  <span
    className="letterify__word"
  >
    <span
      className="letterify__word_char sc-bdVaJa cLaLDe"
    >
      T
    </span>
    <span
      className="letterify__word_char sc-bdVaJa itTzVI"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fqcpQM"
    >
      s
    </span>
    <span
      className="letterify__word_char sc-bdVaJa bhzkWh"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hSnJnP"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cYMYvq"
    >
      l
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cfSahW"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dcDfss"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa xFlAG"
    >
      t
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hqOdAF"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa RkTM"
    >
      r
    </span>
    <span
      className="letterify__word_char sc-bdVaJa gkjZmh"
    >
      i
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hmUMwK"
    >
      f
    </span>
    <span
      className="letterify__word_char sc-bdVaJa lJqJa"
    >
      y
    </span>
    <span
      className="letterify__word_char sc-bdVaJa fWBUwy"
    >
       
    </span>
    <span
      className="letterify__word_char sc-bdVaJa dTaxIf"
    >
      c
    </span>
    <span
      className="letterify__word_char sc-bdVaJa cSZuIy"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jamlIG"
    >
      m
    </span>
    <span
      className="letterify__word_char sc-bdVaJa gvgRZf"
    >
      p
    </span>
    <span
      className="letterify__word_char sc-bdVaJa hTsBGE"
    >
      o
    </span>
    <span
      className="letterify__word_char sc-bdVaJa djNknU"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa eGnfTS"
    >
      e
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jETsEM"
    >
      n
    </span>
    <span
      className="letterify__word_char sc-bdVaJa jbSYzh"
    >
      t
    </span>
  </span>
</div>
`;

exports[`Letterfiy tests missing letterfiy string 1`] = `
<div
  className="no__styleclass"
/>
`;
